src/mon/AuthMonitor.cc

```diff
@@ -1478,8 +1477,8 @@ bool AuthMonitor::prepare_command(MonOpRequestRef op)
       }
     }
 
-    if (!valid_caps(caps_vec, &ss)) {
-      err = -EINVAL;
+    map<string,bufferlist> encoded_caps;
+    if (_check_and_encode_caps(caps_vec, encoded_caps, ss) == -EINVAL) {
```

Comment:
Opps, not setting `err`.

```suggestion
    if (err = _check_and_encode_caps(caps_vec, encoded_caps, ss); err == -EINVAL) {
```